### PR TITLE
Add docs for cubeviz spatial region selection

### PR DESCRIPTION
**Pull request by javerbukh**

### CODE REVIEW

File: docs/cubeviz/displaycubes.rst [hunk 1]
```diff
@@ -45,18 +45,45 @@ visible (checked) or invisible (unchecked).
...
+and the resulting spectrum will be displayed in the 1d spectrum viewer at the bottom of the UI.
+
+.. image:: img/subset_creation.png
+
+There are other options available for region of interest. At the top of the User Interface,
+there is a section that says either "+ No selection (create new)" or "Subset n" where n is an integer
+that tells you which Subset is currently selected. To the right of this area, are red circles that allow
+you to change the method of region selection. The options are `replace`, `add`, `and`, `xor`, and `remove`.
+You can use these options to further adjust the region of interest and adapt it to your use case.
+
```
Comment (PatrickOgle) [inline docs/cubeviz/displaycubes.rst:76]: I would like a description of how to use each option--describe what replace, add, and and xor do when you have them selected and either create a new region or modify an existing one.

**Review by PatrickOgle — CHANGES_REQUESTED** [review]
I think it is important to add some verbiage todescribe how each of the subset logical functions works.

**Review by PatrickOgle — APPROVED** [review]
Looks good.  Thanks for adding this!

File: docs/cubeviz/displaycubes.rst [hunk 2]
```diff
@@ -45,18 +45,50 @@ visible (checked) or invisible (unchecked).
...
+========================
+
+Regions can be created by first clicking the :guilabel:`hammer and screwdriver` icon and
+selecting either the :guilabel:`rectangular region of interest` icon or the
+:guilabel:`circular region of interest` icon, then clicking and dragging (using the mouse) until the
+desired region is covered by a colored shape, after which you release the mouse button.
+Once you have completed the action of clicking and dragging, there is an opportunity to reposition
+the region of interest by clicking on it and moving the region to a different location in the image viewer.
+If you are satisfied with the positioning of your region of interest, simply click another part of the
+viewer to lock it in place. Once a region is selected, the cube will be collapsed in wavelength space over the region,
```
Comment (dtn5ah) [inline docs/cubeviz/displaycubes.rst:66]: > simply click another part of the viewer to lock it in place.

Oh really? You have to actually click somewhere else in order for the moved region to stay? I feel like the user should just be able to let go and it should just work? I'd suggest we should change this behavior later

File: docs/cubeviz/displaycubes.rst [hunk 3]
```diff
@@ -45,18 +45,50 @@ visible (checked) or invisible (unchecked).
...
-=====================================
+Defining Spatial Regions
+========================
+
+Regions can be created by first clicking the :guilabel:`hammer and screwdriver` icon and
+selecting either the :guilabel:`rectangular region of interest` icon or the
+:guilabel:`circular region of interest` icon, then clicking and dragging (using the mouse) until the
+desired region is covered by a colored shape, after which you release the mouse button.
+Once you have completed the action of clicking and dragging, there is an opportunity to reposition
+the region of interest by clicking on it and moving the region to a different location in the image viewer.
```
Comment (dtn5ah) [inline docs/cubeviz/displaycubes.rst:64]:
```suggestion
the region of interest by dragging the region to a different location in the image viewer.
```

I read this as a full KeyOn and KeyOff event, followed by dragging the mouse with no buttons pressed. I thought it was a click-and-drag event in order to move it? If I'm mistaken, feel free to ignore this one!

**Review by dtn5ah — APPROVED** [review]
These suggestions are getting super deep into assumptions and default behaviors I've trained into my fingers that I'm no longer conscious about :/ If what I'm describing isn't actually how it works, feel free to ignore them!

File: docs/cubeviz/displaycubes.rst (hunk 2 above)
Comment (javerbukh) [reply docs/cubeviz/displaycubes.rst:66]: Yes, I think the idea is that you should be able to move the region around before locking it in place, so it does not lock until you click elsewhere in the image viewer.

File: docs/cubeviz/displaycubes.rst (hunk 3 above)
Comment (javerbukh) [reply docs/cubeviz/displaycubes.rst:64]: I'm not exactly sure what the question is. You click and drag to create the region, which is completed once the mouse button is released. The user then has an opportunity to reposition the region by clicking on it once again and moving (dragging) the region to a different location. Once the user clicks off of the region, it is locked.

Comment (javerbukh) [reply docs/cubeviz/displaycubes.rst:64]: I think I understand now, I'll switch the wording to "clicking and dragging the region"

File: docs/cubeviz/displaycubes.rst (hunk 2 above)
Comment (PatrickOgle) [reply docs/cubeviz/displaycubes.rst:66]: The existing behavior allows you to continue editing a region with the selected logical function until you click elsewhere. 
Maybe don't require the click and the region remains editable until another subset is generated?  However, that would
mean autogenerating the collapse after each sub-region edit.

**Pull request merged**